interpreter/cling/lib/MetaProcessor/MetaProcessor.cpp

```diff
@@ -79,10 +85,14 @@ namespace cling {
   }
 
   void MetaProcessor::MaybeRedirectOutputRAII::pop() {
-    if (m_isCurrentlyRedirecting & kSTDOUT) {
+    //If we have only one redirection RAII
+    //only then do the unredirection.
+    if (m_isCurrentlyRedirecting & kSTDOUT
+        && m_MetaProcessor->getRedirectionRAIILevel() == 1) {
```

Comment:
I would move this out of these if-s and instead have right on top of the function:
```
if (m_MetaProcessor->getRedirectionRAIILevel() != 1)
  return;
```